migration/schema/v1/tables.go

```diff
@@ -1011,3 +1011,113 @@ func ConvertV2SamlToIdpConfig(samlMap map[string]any) (v3Config *IdpConfig, err
 	v3Config.Enabled = true
 	return
 }
+
+type Role int
+
+const (
+	Authenticated Role = iota
+	Viewer
+	Auditor
+	Editor
+	Operator
+	Administrator
+	CloudPakAdmin
+	ClusterAdmin
+)
+
+// ToString returns an IAM-compatible role name as a string.
+func (r Role) ToString() (s string) {
+	switch r {
+	case ClusterAdmin:
+		return "ClusterAdministrator"
+	case CloudPakAdmin:
+		return "CloudPakAdministrator"
+	case Administrator:
+		return "Administrator"
+	case Operator:
+		return "Operator"
+	case Editor:
+		return "Editor"
+	case Viewer:
+		return "Viewer"
+	case Auditor:
+		return "Auditor"
+	default:
+		return
+	}
+}
+
+// GetRole takes a string role name and returns the corresponding Role. If the string is not a known name, it returns
+// the lowest role.
+func GetRole(s string) (r Role) {
+	switch s {
+	case "ClusterAdministrator":
+		return ClusterAdmin
+	case "CloudPakAdministrator":
+		return CloudPakAdmin
+	case "Administrator":
+		return Administrator
+	case "Operator":
+		return Operator
+	case "Editor":
+		return Editor
+	case "Viewer":
+		return Viewer
+	case "Auditor":
+		return Auditor
+	default:
+		return Authenticated
+	}
+}
+
+// ToV3String returns an IM-compatible role name as a string. In IM, there are only three valid roles - Administrator,
+// Viewer, and Authenticated.
+func (r Role) ToV3String() (s string) {
+	if r >= Administrator && r <= ClusterAdmin {
+		return Administrator.ToString()
+	} else if r >= Viewer && r < Administrator {
```

Comment:
There are two roles Administrator role and empty role
const CP3_AUTHENTICATED_ROLE = '';